### PR TITLE
🎉 source Salesloft: added new emails incremental stream

### PR DESCRIPTION
**Pull request by yannibenoit**
## What

> Issue: https://github.com/airbytehq/airbyte/issues/11198
> Community Slack Message: https://airbytehq.slack.com/archives/C01VDDEGL7M/p1647435938137639

New source added for Salesloft integration -> [activities/emails.json](https://developers.salesloft.com/api.html#!/Emails/get_v2_activities_emails_json)

image

## How
- Added new schema for this new endpoint -> `emails.json`
- Added new `IncrementalSalesloftStream` class ->  `Emails`
- Edited `streams` method in `SourceSalesloft` class to add new class in return
-  Edited `test_source.py` to allow one more stream when unit testing

## Recommended reading order
1. `emails.json`
2. `source.py`
3.  `test_source.py`

## 🚨 User Impact 🚨

New stream will be available in connector settings

image

New tables will be available in destination

**
image
**

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by alafanechere** [comment]
Thank you @yannibenoit for the contribution. I'll run the acceptance tests and go for a first review asap.

File: airbyte-integrations/connectors/source-salesloft/source_salesloft/schemas/emails.json [hunk 1]
```diff
@@ -0,0 +1,189 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "properties": {
+        "id": {
+            "type": ["null", "integer"]
```
Comment (alafanechere) [inline airbyte-integrations/connectors/source-salesloft/source_salesloft/schemas/emails.json:6]:
```suggestion
            "type":  "integer"
```
Please double-check which fields are nullable, here for example `id` is probably not nullable. The `updated_at` field should also not be nullable if you want to use it as the cursor field. If Salesloft does not provide this information please at lease set the following field as not nullable:
`id`, `created_at`, `updated_at`

**Review by alafanechere — CHANGES_REQUESTED** [review]
Thank you @yannibenoit for this contribution! I made a small suggestion about nullable fields.

File: airbyte-integrations/connectors/source-salesloft/source_salesloft/schemas/emails.json (hunk 1 above)
Comment (alafanechere) [reply airbyte-integrations/connectors/source-salesloft/source_salesloft/schemas/emails.json:6]: Yes please @yannibenoit

Comment (yannibenoit) [reply airbyte-integrations/connectors/source-salesloft/source_salesloft/schemas/emails.json:6]: but @alafanechere we have a similar problem on other schemas 🤔 
id, created_at, updated_at are nullable on the following schemas:
- cadences.json
- people.json
- cadence_memberships.json

Comment (yannibenoit) [reply airbyte-integrations/connectors/source-salesloft/source_salesloft/schemas/emails.json:6]: Made the changes for all schema , let's try out

Comment (alafanechere) [reply airbyte-integrations/connectors/source-salesloft/source_salesloft/schemas/emails.json:6]: Could you please only change your stream's schema to make this PR atomic, and maybe open a second one for other streams?

Comment (alafanechere) [reply airbyte-integrations/connectors/source-salesloft/source_salesloft/schemas/emails.json]: As this PR is aimed at adding a new stream I find it better to not refactor the other streams in the same PR.
If you're working on PR to change the existing stream feel free to change their schema in this new PR too.
Or if you prefer to align all the schemas in a single PR feel free to do it too.

**Review by alafanechere — APPROVED** [review]
Thank you for the changes, publishing now 🎉

**Pull request merged**